src/silx/io/nxdata/_utils.py

```diff
@@ -115,21 +125,25 @@ def get_signal_name(group):
                 if signal_attr in [1, b"1", "1"]:
                     # This is the main (default) signal
                     break
+    # TODO: Fix typing
```

Comment:
Runtime typing sometimes does not agree with the static typing. 

For these cases, I added `TODO` since I would have to change the actual code to fix it and did not want to bloat the PR.